File: C006_CopyTextFile.js
const fs = require('fs');


if (process.argv.length < 4) {
  console.log("Two comand-line arguments are required");
  process.exit(1);
}

var file = fs.readFileSync(process.argv[2], 'utf8');
fs.writeFileSync(process.argv[3], file, function(err) {
    if(err) {
        return console.log(err);
    }

    console.log("The file was saved!");
}); 
// Note since you're using fs.readFile which reads the entire file - it deals with opening and closing file for you, so you don't need to close anything

var lineCount = file.split('\n').length;
console.log("%d lines copied from %s to %s", lineCount, process.argv[2], process.argv[3]);

File: C006_CopyTextFile.py
#!/usr/bin/env python36

import sys

if (len(sys.argv) < 2):
  print ("Two comand-line arguments are required")
  sys.exit(1)

inputFile = open(sys.argv[1], "r")
outputFile = open(sys.argv[2], "w")

lineCount = 0
pos = inputFile.tell()
while ( True ):
  line = inputFile.readline()
  outputFile.write(line)
  lineCount += 1

  newpos = inputFile.tell()
  if newpos == pos:
    break
  else:
    pos = newpos
inputFile.close()
outputFile.close()

print (lineCount, "lines copied from", sys.argv[1], "to", sys.argv[2] )
